### PR TITLE
[T4A7][T13-B1] Low Bao Ling Vivian

### PR DESCRIPTION
**Pull request by sunset1215**
Ready for review

### CODE REVIEW

File: src/seedu/addressbook/data/person/Name.java [hunk 1]
```diff
@@ -61,4 +61,13 @@ public int hashCode() {
         return fullName.hashCode();
     }
 
+    /**
+     * Returns true of the other name is very similar to this name.
+     * Two names are considered similar if ...
+     */
+    public boolean isSimilar(Name other) {
+        // TODO Auto-generated method stub
+        return true;
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java:104]: to stay true to the TDD philosophy, you should start from an empty method ;-)

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 2]
```diff
@@ -9,10 +9,22 @@
 public class NameTest {
 
     @Test
+    public void testConstructor_throwIllegalValueExceptionIfContainNonAlphabeticCharacters() {
+        try {
+            Name name = new Name("1234!!");
+            fail("Should throw IllegalValueException");
+        } catch (IllegalValueException ive) {
```
Comment (okkhoy) [inline test/java/seedu/addressbook/data/person/NameTest.java]: in general you can make tests as `void test_x_y_z throws <someException> {` no need to `try ... catch` in the test

File: test/java/seedu/addressbook/data/person/NameTest.java [hunk 3]
```diff
@@ -9,10 +9,22 @@
 public class NameTest {
 
     @Test
+    public void testConstructor_throwIllegalValueExceptionIfContainNonAlphabeticCharacters() {
```
Comment (okkhoy) [inline test/java/seedu/addressbook/data/person/NameTest.java]: you can eliminate the word `test` from the method name; 
the test name should have 3 parts `<method under test>_<which scenario/test case>_<expected return value>`

File: src/seedu/addressbook/data/person/Name.java [hunk 4]
```diff
@@ -61,4 +62,65 @@ public int hashCode() {
...
+     * Two names are considered similar if: 
+     * <pre>
+     * 1. they are in different case
+     * 2. they are subset/superset of each other
+     * </pre>
+     */
+    public boolean isSimilar(Name other) {
+        if(!isOtherNameNull(other)) {
+            return isNameEqualsIgnoreCase(other) ||
+                 isNameSubsetOfOther(other) ||
```
Comment (okkhoy) [inline src/seedu/addressbook/data/person/Name.java]: think how you can eliminate one the methods by using different objects and the same method!

**Comment by okkhoy** [comment]
@sunset1215 good job on this PR. there are a few areas of improvement which I have pointed out in the comments.
Please take a look.
Please ack & close the PR after reading comments.

**Comment by sunset1215** [comment]
@okkhoy 
Made changes after reading the comments. Please review.

File: src/seedu/addressbook/data/person/Name.java (hunk 1 above)
Comment (sunset1215) [reply src/seedu/addressbook/data/person/Name.java:104]: Noted :-)

**Pull request closed without merging**